tools-in-path/test.sh

```diff
@@ -20,6 +20,6 @@ else
 fi
 
 framework_dir=$(../dotnet-directory --framework "$1")
-dotnet tool update -g dotnet-ef --version "${framework_dir#*App/}.*-*"
+dotnet tool update -g dotnet-ef --version "${framework_dir#*App/}.*"
```

Comment:
I am not sure why we previously used `*-*`. Will this change have a negative impact? Thoughts, @tmds ?

Reply:
With a preview version, this was evaluated as `9.0.0-preview.1.24080.9*-*`, which doesn't match anything. But `9.0.0-preview.1.24080.9*` does.

Reply:
@nicrowe00 updated this test since it was failing for .NET 8.0.1 here: https://github.com/redhat-developer/dotnet-regular-tests/pull/333.

This test is skipped in our CI, so we haven't noticed it doesn't work against preview versions.

With the March release (`8.0.3`), we should be able to go back to how this was before https://github.com/redhat-developer/dotnet-regular-tests/pull/333 because `dotnet-ef` should then run on any 8.0 runtime version.

Reply:
Should I merge this change, then, or defer it to a separate PR (or never)?

Reply:
@nicrowe00 can you remind us why we had to use `*-*` here?

Reply:
> can you remind us why we had to use *-* here?

He didn't add it. It was already there.

Before https://github.com/redhat-developer/dotnet-regular-tests/pull/333 we prefixed this only with the major version number. So the `--version` argument was for example `7.*-*`. This means the SDK it is allowed to install a version higher than 7. It will prefer to install a stable version (e.g. `7.0.0`) but if there is no stable version, it can install a preview (e.g. `7.0.0-preview1`).

With https://github.com/redhat-developer/dotnet-regular-tests/pull/333, if you run against a released version of .NET, you get something like `8.0.1.*-*`. For the SDK this is an acceptable version pattern.

When https://github.com/redhat-developer/dotnet-regular-tests/pull/333 runs against a .NET 9 preview, the `--version` becomes something like  `9.0.0-preview.1.24080.9*-*` and this pattern doesn't make sense to the SDK.

This PR is now changing that version into `9.0.0-preview.1.24080.9.*`. And if this is working, it seems the SDK treats this the same as `9.0.0-*` (since the published `dotnet-ef` version is actually `9.0.0-preview.1.24081.2`).

> Should I merge this change, then, or defer it to a separate PR (or never)?

I'm fine with either what you have here, or the pattern we had before https://github.com/redhat-developer/dotnet-regular-tests/pull/333 (which will work for .NET 9 previews, and should start working for .NET 8 on the next patch release).

Reply:
> He didn't add it. It was already there.

I was thinking of #319.

Reply:
Because I suggested it in that PR for the reasons mentioned above.

Reply:
This is the documentation for the wildcard pattern: https://github.com/NuGet/Home/wiki/Support-pre-release-packages-with-floating-versions.